docs/egress.md

```diff
@@ -198,6 +199,52 @@ The `ipRanges` field contains a list of IP ranges representing the available IPs
 of this IP pool. Each IP range may consist of a `cidr` or a pair of `start` and
 `end` IPs (which are themselves included in the range).
 
+### SubnetInfo
+
+By default, it's assumed that the IPs allocated from the pool are in the same
+subnet as the Node IPs. Starting with Antrea v1.15, IPs can be allocated from a
+subnet different from the Node IPs.
+
+The optional `subnetInfo` field contains the subnet attributes of the IPs in
+this pool. When using a different subnet:
+
+* `gateway` and `prefixLength` must be set. Antrea will route Egress traffic to
+the specified gateway when the destination is not in the same subnet of the
+Egress IP, otherwise route it to the destination directly.
+
+* Optionally, you can specify `vlan` if the underlying network is expecting it.
+Once set, Antrea will tag Egress traffic leaving the Egress Node with the
+specified VLAN ID. Correspondingly, it's expected that reply traffic towards
+these Egress IPs is also tagged with the specified VLAN ID when arriving at the
+Egress Node.
+
+An example of ExternalIPPool using a different subnet is as below:
+
+```yaml
+apiVersion: crd.antrea.io/v1beta1
+kind: ExternalIPPool
+metadata:
+  name: prod-external-ip-pool
+spec:
+  ipRanges:
+  - start: 10.10.0.2
+    end: 10.10.0.10
+  subnetInfo:
+    gateway: 10.10.0.1
+    prefixLength: 24
+    vlan: 10
+  nodeSelector:
+    matchLabels:
+      network-role: egress-gateway
+```
+
+**Note**: Specifying different subnets is currently in alpha version. To use
+this feature, users should enable the `EgressSeparateSubnet` feature gate.
+Currently, the maximum number of different subnets that can be supported in a
+cluster is 20, which should be sufficient for most cases. If you need to have
```

Comment:
The limit is per Node, not per cluster?

Reply:
It's per Node, however it will be risky for users to use more than 20 subnets in a cluster as they don't have control over how the Egress IPs are distributed, and in worst case (like most Egress candidate Nodes crash and Egress are scheduled to a few remaining Nodes) it may sometimes work and sometimes not. So I feel we should just ask users to not use 20 subnets in a cluster before we support it.

Another reason is if say 20 subnets per Node, we need to elaborate they can't assume the cluster limit is not 20*Nodes...

Reply:
Ok, we can start from 20 per cluster. But I assume there can be cases that users configure different VLANs on different sets of Egress Nodes. This seems more likely to happen when Nodes are connected to physical network. We can see if users ask for more VLANs.

Reply:
sure